Add logging to the distro (#336)
signalfx/splunk-otel-go: distro/README.md

forward all telemetry to Splunk. See [`example_test.go`](./example_test.go) for
a complete example.

## Logging

> By default the [`SDK`] will log relevant operation information, warnings, and
> errors. This logging can be explicitly configured in the code that creates the
> [`SDK`] using the `WithLogger` option to configure a logger that only logs what
> is wanted. Or, the default logging level can be set using the OpenTelemetry
> defined `OTEL_LOG_LEVEL` environment variable. The following values for that
> variable are understood.

Reviewer: ```suggestion
By default, the [`SDK`] logs relevant information, warnings, and errors. 
You can set the default logging level by setting the `OTEL_LOG_LEVEL` 
environment variable to one of the following values:
```

> - `"error"`: only log errors
> - `"warn"`: log errors and warnings
> - `"info"`: log operation information, warnings, and errors
> - `"debug"`: log debugging and operation information, warnings, and errors
> 

Reviewer: ```suggestion
- `"error"`: Log errors
- `"warn"`: Log errors and warnings
- `"info"`: Log information, warnings, and errors
- `"debug"`: Log debugging and operation information, warnings, and errors

Logging can be explicitly configured in the code that creates the [`SDK`] 
using the `WithLogger` option.
```

## Configuration

> The [`SDK`] is configured with the following options.

Reviewer: We usually don't mention the SDK in other distros's repos. I assume this just describes our distribution defaults?

Author: Yes, this specifically describes option types that only our distro accepts and the unique defaults we have.

- Default value: none; an non-TLS connection is used.
- Environment variable: n/a (only configurable in code).

### `WithLogger`

`WithLogger` configures the logger used by this distro.

> The [`logr.Logger`] provided should be configured with a verbosity enabled to
> emit [`Info`] logs of the desired level. The following log level to verbosity
> value are used.

Reviewer: ```suggestion
Configure [`logr.Logger`] with a verbosity value that allows to emit
 [`Info`] logs of the desired level. The SDK uses the following log level 
 to verbosity value correspondences:
```

> - warning: `0`
> - info: `1`
> - debug: `2+`

Reviewer: ```suggestion
| Log level | Verbosity value |
|---|---|
| warning | `0` |
| info | `1` |
| debug | `2` or higher |
```

> - Default value: [`zapr`]; configured for info logging.
> - Environment variable: n/a (only configurable in code).

Reviewer: ```suggestion
The default value, [`zapr`], is configured for `info` logging. This setting is configurable only in code.
```
